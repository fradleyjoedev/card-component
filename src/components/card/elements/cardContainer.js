import styled from 'styled-components'

export default styled.div`
    background-image: url('${props => props.image}');
`;