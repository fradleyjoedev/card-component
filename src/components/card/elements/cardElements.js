import styled from 'styled-components'

//Made the assumption that the image should cover the container
export const CardContainer = styled.div`
    background-image: url('${props => props.image}');
    background-size: cover;
    display: flex;
    flex-wrap: wrap;
    align-content: flex-end;
    width: 100%;
    height: 100%;
`;