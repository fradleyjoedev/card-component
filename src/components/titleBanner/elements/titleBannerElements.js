import styled from 'styled-components'
import { COLOUR_PRIMARY, COLOUR_SECONDARY } from '../../shared/globalColors';
import { MEDIAQUERY_TABLET } from '../../shared/globalMediaQueries';

export const TitleBannerContainer = styled.div`
    display: flex;
    align-items: center;
    width: 100%
    height: 20%;
    background: ${COLOUR_SECONDARY};
    padding: 2% 1%;
`;

export const SmallImage = styled.span`
    background-image: url('${props => props.image}');
    height: 100%;
    width: 20%;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
`;

export const Heading = styled.h2`
    color: ${COLOUR_PRIMARY};
    font-size: 5.2vw;
    width: 80%;
    padding-left: 2%;

    @media (min-width: ${MEDIAQUERY_TABLET}) {
        font-size: 3.2vw;
    }
`;