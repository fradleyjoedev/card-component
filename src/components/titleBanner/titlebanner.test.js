import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components'
import TitleBanner from '.';
import { SmallImage } from './elements/titleBannerElements';

describe('TitleBanner Component', () => {
  it('should render correctly', () => {
    const json = {
        "image": "image",
        "heading": "test"
    };
    const component = shallow(<TitleBanner data={json} />);
    
    expect(component).toMatchSnapshot();
  });

  it('should have a background image', () => {
    const imageURL = "image.jpg";

    const component = mount(<SmallImage image={imageURL}/>);
    
    expect(component).toHaveStyleRule('background-image', `url('${imageURL}')`);
  });
});