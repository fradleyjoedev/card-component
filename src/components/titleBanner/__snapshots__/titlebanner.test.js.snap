// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBanner Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TitleBanner
    data={
      Object {
        "heading": "test",
        "image": "image",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          image="image"
        />,
        <ForwardRef>
          test
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "image": "image",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
    background-image: url('",
              [Function],
              "');
    height: 100%;
    width: 20%;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    color: ",
              "#ffffff",
              ";
    font-size: 5.2vw;
    width: 80%;
    padding-left: 2%;

    @media (min-width: 768px) {
        font-size: 3.2vw;
    }
",
            ],
          },
          "displayName": "styled.h2",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "h2",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    align-items: center;
    width: 100%
    height: 20%;
    background: ",
          "#000000ba",
          ";
    padding: 2% 1%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            image="image"
          />,
          <ForwardRef>
            test
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "image": "image",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
    background-image: url('",
                [Function],
                "');
    height: 100%;
    width: 20%;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    color: ",
                "#ffffff",
                ";
    font-size: 5.2vw;
    width: 80%;
    padding-left: 2%;

    @media (min-width: 768px) {
        font-size: 3.2vw;
    }
",
              ],
            },
            "displayName": "styled.h2",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "h2",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    align-items: center;
    width: 100%
    height: 20%;
    background: ",
            "#000000ba",
            ";
    padding: 2% 1%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
